plugins/xdebug/xdebug_headers.h

```diff
@@ -0,0 +1,83 @@
+/** @file
+ *
+ * XDebug plugin headers functionality declarations.
+ *
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#pragma once
+
+#include <sstream>
+#include "ts/ts.h"
+
+namespace xdebug
+{
+
+/**
+ * Whether to include whitespace in the output.
+ */
+static constexpr bool JSON_COMPLIANT = true;
```

Comment:
This constant and having `json_compliant` passed around is confusing.  Why do we need this? What determines when it should be true? Why is sometimes a variable passed and sometimes hardcoded to this?  Some functions have a flag and some have a `_json` suffix.